lib/general/network/api_manager.dart

```diff
@@ -0,0 +1,38 @@
+import 'dart:async';
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:http/http.dart' as http;
+
+import 'custom_exceptions.dart';
+
+/// Manage API calls for simple http library.
+/// More sophisticated http library should support error handling with its own exceptions.
+class ApiManager {
+  Future<dynamic> getAPICall(Uri url) async {
+    print("Calling API: $url"); // todo logger
+
+    var jsonResponse;
+    try {
+      final httpResponse = await http.get(url);
+      jsonResponse = _validateResponse(httpResponse);
+    } on SocketException {
+      throw NoInternetConnectionException();
+    }
+    return jsonResponse;
+  }
+
+  dynamic _validateResponse(http.Response response) async {
```

Comment:
no need to be async since there is no await keyword

Reply:
Updated here https://github.com/Swirastlynn/skitour-planning/tree/bloc-architecture